### PR TITLE
feat: add tutorial 

### LINKED ISSUE #261: Create Tutorial Section
### Subject

Tutorial

### Description

I'd love for us to have a tutorial section in the docs similar to: https://www.solidjs.com/tutorial/introduction_basics

We can leverage Sandpack by Codesandbox for the code and preview area (or figure out how Solid.js did that).

We're going to have different sections and under each section, we need practical step-by-step tutorials to guide the newbie users

- Basics (Style Props)
- Pseudo Props
- Layout components
- Interactive components
- Theming Basics (tokens and semantic tokens)
- Theming Advanced (component theming)

### PR DESCRIPTION
**Pull request by noobinthisgame**
Closes #261 

## 📝 Description

This PR creates a tutorial section with an interactive codesanbox to directly follow along with it. 

Open tasks for MVP:
- [x] Fix file update with smaller code samples
- [x] Fix build
- [x] Refactor code

## 📝 Additional Information

In the future there can be some improvements made to QOL like autocomplete

There is no link atm to reach the tutorial. Use /tutorial/basics as entry point :)

### CODE REVIEW

**Comment by noobinthisgame** [comment]
@nikolovlazar This is ready for review 🎉

File: configs/sandpack-contents/tutorial/package.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "main": "src/index.js",
+  "dependencies": {
+    "@chakra-ui/react": "1.8.6",
+    "@emotion/react": "11.8.1",
+    "@emotion/styled": "11.8.1",
+    "framer-motion": "6.2.8",
+    "hash-it": "5.0.2",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
+    "react-jazzicon": "1.0.3",
```
Comment (nikolovlazar) [inline configs/sandpack-contents/tutorial/package.ts:19]: Where's the `react-jazzicon` package being used?

File: layouts/mdx.tsx [hunk 2]
```diff
@@ -31,11 +33,19 @@ interface MDXLayoutProps {
...
+  // const route = findRouteByPath(removeFromLast(frontmatter.slug, '#'), routes)
+  // const routeContext = getRouteContext(route, routes)
+  const routes = getRoutes(frontmatter.slug)
+  console.log('routes', routes)
+  console.log('removeLast', removeFromLast(frontmatter.slug, '#'))
   const route = findRouteByPath(removeFromLast(frontmatter.slug, '#'), routes)
+  console.log('route', route)
   const routeContext = getRouteContext(route, routes)
 
+  console.log('routeContext', routeContext)
```
Comment (nikolovlazar) [inline layouts/mdx.tsx:47]: Let's clean up the unused commented code and console logs.

File: layouts/tutorial.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  frontmatter: any
+  children: ReactNode
+}
+
+const TutorialMenu = ({
+  headings,
+  routes,
+}: {
+  headings: any[]
+  routes: any[]
```
Comment (nikolovlazar) [inline layouts/tutorial.tsx:31]: Can we make types for these?

File: src/components/footer.tsx [hunk 4]
```diff
@@ -72,8 +80,8 @@ const NigeriaFlag = (props) => (
   </chakra.svg>
 )
 
-export const Footer = () => (
-  <VStack as='footer' spacing={4} mt={12} textAlign='center'>
+export const Footer: React.FunctionComponent<StackProps> = (props) => (
```
Comment (nikolovlazar) [inline src/components/footer.tsx:83]: Why are we setting the `Footer` type to `React.FunctionalComponent` in this case?

File: src/components/header.tsx [hunk 5]
```diff
@@ -154,7 +155,7 @@ function Header(props: HTMLChakraProps<'header'>) {
       width='full'
       {...props}
     >
-      <chakra.div height='4.5rem' mx='auto' maxW='8xl'>
+      <chakra.div height='4.5rem' mx='auto' maxW={maxW} maxWidth={maxWidth}>
```
Comment (nikolovlazar) [inline src/components/header.tsx:158]: Double max width props?

File: src/components/sidebar/sidebar.tsx [hunk 6]
```diff
@@ -119,7 +119,7 @@ export const isMainNavLinkActive = (href: string, path: string) => {
   )
 }
 
-const MainNavLink = ({ href, icon, children }: MainNavLinkProps) => {
+export const MainNavLink = ({ href, icon, children }: MainNavLinkProps) => {
```
Comment (nikolovlazar) [inline src/components/sidebar/sidebar.tsx:122]: Do we need to export these? I don't see them being used anywhere.

**Review by nikolovlazar — CHANGES_REQUESTED** [review]
Great work on this, @noobinthisgame! I've left some comments for you. Not all of them are strict changes, but discussions.

File: layouts/mdx.tsx (hunk 2 above)
Comment (noobinthisgame) [reply layouts/mdx.tsx:47]: Good catch ! Will do :)

File: configs/sandpack-contents/tutorial/package.ts (hunk 1 above)
Comment (noobinthisgame) [reply configs/sandpack-contents/tutorial/package.ts:19]: This was in the initial example for the csb. I'll check and clean it up.

File: layouts/tutorial.tsx (hunk 3 above)
Comment (noobinthisgame) [reply layouts/tutorial.tsx:31]: I'll see what I can do :)

File: src/components/footer.tsx (hunk 4 above)
Comment (noobinthisgame) [reply src/components/footer.tsx:83]: Looks like a relict from some trials I had. :D

File: src/components/sidebar/sidebar.tsx (hunk 6 above)
Comment (noobinthisgame) [reply src/components/sidebar/sidebar.tsx:122]: Relict from first draft, will remove.

File: src/components/header.tsx (hunk 5 above)
Comment (noobinthisgame) [reply src/components/header.tsx:158]: As I needed to pass a maxWidth to this for the tutorial Layout, I've built it like this. People can then use either the short `maxW` or `maxWidth` to achieve this.

**Comment by noobinthisgame** [comment]
@nikolovlazar I've applied some changes for your comments :)

**Review by nikolovlazar — APPROVED** [review]
LGTM! 🚀

**Pull request merged**